### ISSUE TITLE
other languages

### ISSUE THREAD

**Issue by Storyinsea** [ekwek1/soprano#2]
hello，support other languages?

**Comment by DevopsDood** [comment]
I'm down to help add other languages

**Comment by ekwek1** [comment]
Hi everyone, I appreciate all your support for the project! The only limiting factor for multilingual capabilities are high-quality datasets, so please let me know of any datasets for specific languages you would like supported.

**Comment by FGRibreau** [comment]
Hi Eugene!

For French support, here are some high-quality open-source datasets from Mozilla:
- https://datacollective.mozillafoundation.org/datasets/cmj8u3p3x008hnxxb07gm9ke3
- https://datacollective.mozillafoundation.org/datasets/cmj8u48ad0025nxzp2lfaluce
- and from MLS: https://dl.fbaipublicfiles.com/mls/mls_french.tar.gz

Can what to see what soprano can do in French!!!

**Comment by ekwek1** [comment]
@Tribuzeus @FGRibreau Thank you so much! I'll include these in the training data.

@C00reNUT Yes, roughly 15-20 seconds. Of course, since Soprano splits input text by sentence and generates them independently, this means it can support infinite generation, as long as each individual sentence follows the length limit.

**Comment by vinodkrishna221** [comment]
@ekwek1 for indian languages
https://github.com/AI4Bharat/IndicVoices-R/blob/master/data_links.txt check out this it has almost major indian languages with 1700 hours of data

**Comment by karayakar** [comment]
@ekwek1  Started training Turkish language, I would say tokenizer needs to be aligned , added new Turkish chars, clean_text needs to be modified, train script needs to be aligned accordingly. I have done all those, initial 100K step not enough to get clear voice. 
I have 100+Hours clean aligned synthetic voice data.
 
adjusted learning rate and 300K configured, right now it is continuing.. Will let you know the result. 

Good work and thanks for sharing.

**Comment by karayakar** [comment]
Bad news --- 
In short it is not designed to support Globally !! Decoder is custom and female voice  sticked into it.
With the provided scripts it is impossible to train or fine tune new langauge!
 Here is the comparison between the original (default) settings and our modified values:

1. Decoder Architecture (decoder.py)
Parameter	Original (Repo Default)	Modified (Current)	Notes
decoder_dim	512	768	CRITICAL CHANGE. The original code expected 512, but the actual checkpoint file (decoder.pth from HF) is 768. We changed this to match the file.
input_kernel	1	1	Unchanged.
dw_kernel	3	3	Unchanged.
2. Training Configuration (train.py)
Parameter	Original (Repo Default)	Modified (Current)	Notes
text_factor	0.0 (or Implicit)	0.5	Increased to force the model to learn text-audio alignment for the new language.
max_lr	5e-4	1e-4	Decreased to prevent catastrophic forgetting and maintain stability.
grad_accum	1	1	Kept at 1 for frequent updates.
tokenizer	Local Variable	Global Variable	WINDOWS FIX. Moved to global scope to prevent multiprocessing pickling errors on Windows.
resize_token_embeddings	False (Implicit)	True	Added explicitly to support the extended vocabulary (Turkish characters).
save_freq	5000	1000	Increased frequency for safer checkpointing.
3. Data Loading (generate_dataset.py)
Feature	Original	Modified	Notes
Audio Format	Float32 / WAV	Int16 Handling	Added logic to handle int16 wav files by normalizing them to [-1.0, 1.0] range.
Metadata	metadata.csv only	metadata.txt & `	`
4. Text Normalization (soprano/utils/text_normalizer.py)
Feature	Original	Modified	Notes
Unidecode	Enabled	Disabled	Removed unidecode to preserve Turkish characters (e.g., ş -> s, ğ -> g conversion was blocked).
Regex	[a-z...]	Extended	Updated regex to allow Turkish characters in the cleaning process.
Summary of the Conflict
The main technical conflict right now is in the Decoder:

The Code originally wanted a 512-dim decoder.
The Official Checkpoint provided is 768-dim.
The LLM Output (your trained model) is 512-dim.

**Comment by haqatak** [comment]
768 sounds like soprano 1.1
soprano 1.0 is 512

**Comment by karayakar** [comment]
> 768 sounds like soprano 1.1 soprano 1.0 is 512

That is an easy fix, I am able run original model locally... but when you try fine tune or try to train new language it is impossible.


<img width="1631" height="1249" alt="Image" src="https://github.com/user-attachments/assets/48645d00-2f6b-4ef1-81d6-9b516ab6ebe1" />

**Comment by haqatak** [comment]
I just cleaned up a norwegian dataset - I wasn't aware of the 1.0 / 1.1 so got too only garbage results.
if I managed to fine-tune Norwegian I will share the recipe

**Comment by karayakar** [comment]
Decoder is locked, don't waste time on it! Unless proper training code released!
